xdsl/dialects/experimental/fir.py

```diff
@@ -22,21 +35,94 @@
     VarOperand,
     VarOpResult,
     VarRegion,
-    attr_def,
     irdl_attr_definition,
     irdl_op_definition,
     operand_def,
-    opt_attr_def,
     opt_operand_def,
+    opt_prop_def,
```

Comment:
I would recommend against `opt_prop_def` in general, I don't think MLIR has an equivalent, the equivalent is `MyAttr | NoneAttr` as opposed to Python's `None`

Reply:
Is there an example of using prop_def for optional properties in the dialects? I ask because the following fails validation:

`linkName: StringAttr | NoneAttr = prop_def(StringAttr)
linkName: StringAttr  = prop_def(StringAttr | NoneAttr)
linkName: StringAttr | NoneAttr = prop_def(StringAttr | NoneAttr)`

Whereas opt_prop_def obviously does work for this optional property

Reply:
My bad, I don't think we have the necessary functionality to support this right now, we'd first need default attributes